File: app/controllers/User/group.go
// +----------------------------------------------------------------------
// | GoCMS 0.1
// +----------------------------------------------------------------------
// | Copyright (c) 2013-2014 http://www.6574.com.cn All rights reserved.
// +----------------------------------------------------------------------
// | Licensed ( http://www.apache.org/licenses/LICENSE-2.0 )
// +----------------------------------------------------------------------
// | Author: zzdboy <zzdboy1616@163.com>
// +----------------------------------------------------------------------

package controllers

//会员组管理
import "strconv"
import "admin/app/models"
import "github.com/revel/revel"

type Group struct {
	*revel.Controller
}

//首页
func (c *Group) Index(user_group *models.User_Group) revel.Result {
	title := "会员组首页--GoCMS管理系统"

	group_list := user_group.GetGroupList()

	c.Render(title, group_list)
	return c.RenderTemplate("User/Group/Index.html")
}

//添加会员组
func (c *Group) Add(user_group *models.User_Group) revel.Result {

	if c.Request.Method == "GET" {

		title := "添加会员组--GoCMS管理系统"

		c.Render(title)
		return c.RenderTemplate("User/Group/Add.html")
	} else {

		data := make(map[string]string)

		var name string = c.Params.Get("name")
		if len(name) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入会员组名称!"
			return c.RenderJson(data)
		} else {
			user_group.Name = name
		}

		var point string = c.Params.Get("point")
		if len(point) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入积分!"
			return c.RenderJson(data)
		} else {
			Point, err := strconv.ParseInt(point, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Point = Point
		}

		var star string = c.Params.Get("star")
		if len(star) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入星星数!"
			return c.RenderJson(data)
		} else {
			Star, err := strconv.ParseInt(star, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Star = Star
		}

		var allowpost string = c.Params.Get("allowpost")
		Allowpost, err := strconv.ParseInt(allowpost, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowpost = Allowpost

		var allowpostverify string = c.Params.Get("allowpostverify")
		Allowpostverify, err := strconv.ParseInt(allowpostverify, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowpostverify = Allowpostverify

		var allowupgrade string = c.Params.Get("allowupgrade")
		Allowupgrade, err := strconv.ParseInt(allowupgrade, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowupgrade = Allowupgrade

		var allowsendmessage string = c.Params.Get("allowsendmessage")
		Allowsendmessage, err := strconv.ParseInt(allowsendmessage, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowsendmessage = Allowsendmessage

		var allowattachment string = c.Params.Get("allowattachment")
		Allowattachment, err := strconv.ParseInt(allowattachment, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowattachment = Allowattachment

		var allowsearch string = c.Params.Get("allowsearch")
		Allowsearch, err := strconv.ParseInt(allowsearch, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowsearch = Allowsearch

		var priceday string = c.Params.Get("priceday")
		Priceday, err := strconv.ParseFloat(priceday, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Priceday = Priceday

		var pricemonth string = c.Params.Get("pricemonth")
		Pricemonth, err := strconv.ParseFloat(pricemonth, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Pricemonth = Pricemonth

		var priceyear string = c.Params.Get("priceyear")
		Priceyear, err := strconv.ParseFloat(priceyear, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Priceyear = Priceyear

		var allowmessage string = c.Params.Get("allowmessage")
		if len(allowmessage) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入最大短消息数!"
			return c.RenderJson(data)
		} else {
			Allowmessage, err := strconv.ParseInt(allowmessage, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Allowmessage = Allowmessage
		}

		var allowpostnum string = c.Params.Get("allowpostnum")
		if len(allowpostnum) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入日最大投稿数!"
			return c.RenderJson(data)
		} else {
			Allowpostnum, err := strconv.ParseInt(allowpostnum, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Allowpostnum = Allowpostnum
		}

		var usernamecolor string = c.Params.Get("usernamecolor")
		user_group.Usernamecolor = usernamecolor

		var icon string = c.Params.Get("icon")
		user_group.Icon = icon

		var desc string = c.Params.Get("desc")
		user_group.Desc = desc

		var status string = c.Params.Get("status")
		Status, err := strconv.ParseInt(status, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Status = Status

		if user_group.Save() {
			data["status"] = "1"
			data["message"] = "添加成功!"
			return c.RenderJson(data)
		} else {
			data["status"] = "0"
			data["message"] = "添加失败!"
			return c.RenderJson(data)
		}
	}
}

//编辑会员组
func (c *Group) Edit(user_group *models.User_Group) revel.Result {

	if c.Request.Method == "GET" {

		title := "编辑会员组--GoCMS管理系统"

		var id string = c.Params.Get("id")

		if len(id) > 0 {
			Id, err := strconv.ParseInt(id, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}

			user_group_info := user_group.GetById(Id)

			c.Render(title, user_group_info)
		} else {
			c.Render(title)
		}

		return c.RenderTemplate("User/Group/Edit.html")
	} else {

		data := make(map[string]string)

		var id string = c.Params.Get("id")
		if len(id) <= 0 {
			data["status"] = "0"
			data["message"] = "参数错误，编辑失败!"
			return c.RenderJson(data)
		}

		Id, err := strconv.ParseInt(id, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}

		var name string = c.Params.Get("name")
		if len(name) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入会员组名称!"
			return c.RenderJson(data)
		} else {
			user_group.Name = name
		}

		var point string = c.Params.Get("point")
		if len(point) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入积分!"
			return c.RenderJson(data)
		} else {
			Point, err := strconv.ParseInt(point, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Point = Point
		}

		var star string = c.Params.Get("star")
		if len(star) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入星星数!"
			return c.RenderJson(data)
		} else {
			Star, err := strconv.ParseInt(star, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Star = Star
		}

		var allowpost string = c.Params.Get("allowpost")
		Allowpost, err := strconv.ParseInt(allowpost, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowpost = Allowpost

		var allowpostverify string = c.Params.Get("allowpostverify")
		Allowpostverify, err := strconv.ParseInt(allowpostverify, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowpostverify = Allowpostverify

		var allowupgrade string = c.Params.Get("allowupgrade")
		Allowupgrade, err := strconv.ParseInt(allowupgrade, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowupgrade = Allowupgrade

		var allowsendmessage string = c.Params.Get("allowsendmessage")
		Allowsendmessage, err := strconv.ParseInt(allowsendmessage, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowsendmessage = Allowsendmessage

		var allowattachment string = c.Params.Get("allowattachment")
		Allowattachment, err := strconv.ParseInt(allowattachment, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowattachment = Allowattachment

		var allowsearch string = c.Params.Get("allowsearch")
		Allowsearch, err := strconv.ParseInt(allowsearch, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Allowsearch = Allowsearch

		var priceday string = c.Params.Get("priceday")
		Priceday, err := strconv.ParseFloat(priceday, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Priceday = Priceday

		var pricemonth string = c.Params.Get("pricemonth")
		Pricemonth, err := strconv.ParseFloat(pricemonth, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Pricemonth = Pricemonth

		var priceyear string = c.Params.Get("priceyear")
		Priceyear, err := strconv.ParseFloat(priceyear, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Priceyear = Priceyear

		var allowmessage string = c.Params.Get("allowmessage")
		if len(allowmessage) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入最大短消息数!"
			return c.RenderJson(data)
		} else {
			Allowmessage, err := strconv.ParseInt(allowmessage, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Allowmessage = Allowmessage
		}

		var allowpostnum string = c.Params.Get("allowpostnum")
		if len(allowpostnum) <= 0 {
			data["status"] = "0"
			data["message"] = "请输入日最大投稿数!"
			return c.RenderJson(data)
		} else {
			Allowpostnum, err := strconv.ParseInt(allowpostnum, 10, 64)
			if err != nil {
				revel.WARN.Println(err)
			}
			user_group.Allowpostnum = Allowpostnum
		}

		var usernamecolor string = c.Params.Get("usernamecolor")
		user_group.Usernamecolor = usernamecolor

		var icon string = c.Params.Get("icon")
		user_group.Icon = icon

		var desc string = c.Params.Get("desc")
		user_group.Desc = desc

		var status string = c.Params.Get("status")
		Status, err := strconv.ParseInt(status, 10, 64)
		if err != nil {
			revel.WARN.Println(err)
		}
		user_group.Status = Status

		if user_group.Edit(Id) {
			data["status"] = "1"
			data["message"] = "编辑成功!"
			return c.RenderJson(data)
		} else {
			data["status"] = "0"
			data["message"] = "编辑失败!"
			return c.RenderJson(data)
		}
	}
}

File: public/js/User/group.js
// +----------------------------------------------------------------------
// | GoCMS 0.1
// +----------------------------------------------------------------------
// | Copyright (c) 2013-2014 http://www.6574.com.cn All rights reserved.
// +----------------------------------------------------------------------
// | Licensed ( http://www.apache.org/licenses/LICENSE-2.0 )
// +----------------------------------------------------------------------
// | Author: zzdboy <zzdboy1616@163.com>
// +----------------------------------------------------------------------

/**
 * 会员组管理
 */


//添加会员组
function add() {
	art.dialog.open('/Group/add/', {
		id : 'group_add',
		title : '添加用户组',
		width : 880,
		height : 500,
		lock : true,
		ok : function() {
			var iframe = this.iframe.contentWindow;

			var par = [];

			var name = iframe.$('#name').val();
			if (name == '') {
				iframe.$('#name').addClass("onFocus").focus();
				iframe.$('#nametip').addClass("onError").html('请输入会员组名称!');
				return false;
			}else{
				iframe.$('#name').removeClass("onFocus");
				iframe.$('#nametip').removeClass("onError").addClass("onCorrect").html('输入正确');
				var pars = "name=" + name;
				par.push(pars);
			}

			var point = iframe.$('#point').val();
			if (point == '') {
				iframe.$('#point').addClass("onFocus").focus();
				iframe.$('#pointtip').addClass("onError").html('请输入积分!');
				return false;
			}else{
				iframe.$('#point').removeClass("onFocus");
				iframe.$('#pointtip').removeClass("onError").addClass("onCorrect").html('输入正确');
				var pars = "point=" + point;
				par.push(pars);
			}

			var star = iframe.$('#star').val();
			if (star == '') {
				iframe.$('#star').addClass("onFocus").focus();
				iframe.$('#startip').addClass("onError").html('请输入星星数!');
				return false;
			}else{
				iframe.$('#star').removeClass("onFocus");
				iframe.$('#startip').removeClass("onError").addClass("onCorrect").html('输入正确');
				var pars = "star=" + star;
				par.push(pars);
			}

			if (iframe.$('#allowpost').is(":checked")) {
				var pars = "allowpost=1";
				par.push(pars);
			}else{
				var pars = "allowpost=0";
				par.push(pars);
			}

			if (iframe.$('#allowpostverify').is(":checked")) {
				var pars = "allowpostverify=1";
				par.push(pars);
			}else{
				var pars = "allowpostverify=0";
				par.push(pars);
			}

			if (iframe.$('#allowupgrade').is(":checked")) {
				var pars = "allowupgrade=1";
				par.push(pars);
			}else{
				var pars = "allowupgrade=0";
				par.push(pars);
			}

			if (iframe.$('#allowsendmessage').is(":checked")) {
				var pars = "allowsendmessage=1";
				par.push(pars);
			}else{
				var pars = "allowsendmessage=0";
				par.push(pars);
			}

			if (iframe.$('#allowattachment').is(":checked")) {
				var pars = "allowattachment=1";
				par.push(pars);
			}else{
				var pars = "allowattachment=0";
				par.push(pars);
			}

			if (iframe.$('#allowsearch').is(":checked")) {
				var pars = "allowsearch=1";
				par.push(pars);
			}else{
				var pars = "allowsearch=0";
				par.push(pars);
			}

			var priceday = iframe.$('#priceday').val();
			if (priceday == '') {
				var pars = "priceday=0";
				par.push(pars);
			}else{
				var pars = "priceday=" + priceday;
				par.push(pars);
			}

			var pricemonth = iframe.$('#pricemonth').val();
			if (pricemonth == '') {
				var pars = "pricemonth=0";
				par.push(pars);
			}else{
				var pars = "pricemonth=" + pricemonth;
				par.push(pars);
			}

			var priceyear = iframe.$('#priceyear').val();
			if (priceyear == '') {
				var pars = "priceyear=0";
				par.push(pars);
			}else{
				var pars = "priceyear=" + priceyear;
				par.push(pars);
			}

			var allowmessage = iframe.$('#allowmessage').val();
			if (allowmessage == '') {
				iframe.$('#allowmessage').addClass("onFocus").focus();
				iframe.$('#allowmessagetip').addClass("onError").html('请输入最大短消息数!');
				return false;
			}else{
				iframe.$('#allowmessage').removeClass("onFocus");
				var pars = "allowmessage=" + allowmessage;
				par.push(pars);
			}

			var allowpostnum = iframe.$('#allowpostnum').val();
			if (allowpostnum == '') {
				iframe.$('#allowpostnum').addClass("onFocus").focus();
				iframe.$('#allowpostnumtip').addClass("onError").html('请输入日最大投稿数!');
				return false;
			}else{
				iframe.$('#allowpostnum').removeClass("onFocus");
				var pars = "allowpostnum=" + allowpostnum;
				par.push(pars);
			}

			var usernamecolor = iframe.$('#usernamecolor').val();
			var pars = "usernamecolor=" + usernamecolor;
			par.push(pars);

			var icon = iframe.$('#icon').val();
			var pars = "icon=" + icon;
			par.push(pars);

			var desc = iframe.$('#desc').val();
			var pars = "desc=" + desc;
			par.push(pars);

			var status = iframe.$("input[name='status']:checked").val();
			var pars = "status=" + status;
			par.push(pars);

			pars = par.join("&");

			$.ajax({
				type : "POST",
				url : "/Group/add/",
				data : pars,
				success : function(tmp) {
					if (tmp.status == 1) {
						right_refresh();
						notice_tips("添加成功!");
					} else {
						notice_tips(tmp.message);
					}
				}
			});
		},
		cancel : true
	});
}

/**
 * 编辑用户组
 */
function edit(id) {
	if (id == '') {
		notice_tips("参数错误!");
		return false;
	}

	art.dialog.open('/Group/edit/' + id + '/', {
		id : 'group_edit',
		title : '编辑用户',
		width : 880,
		height : 500,
		lock : true,
		ok : function() {
			var iframe = this.iframe.contentWindow;

			var par = [];

			var pars = "id=" + id;
			par.push(pars);

			var name = iframe.$('#name').val();
			if (name == '') {
				iframe.$('#nametip').removeClass("onShow").addClass("onError").html('请输入会员组名称!');
				return false;
			}else{
				iframe.$('#nametip').removeClass("onError").addClass("onCorrect").html('正确');
				var pars = "name=" + name;
				par.push(pars);
			}

			var point = iframe.$('#point').val();
			if (point == '') {
				iframe.$('#pointtip').removeClass("onShow").addClass("onError").html('请输入积分!');
				return false;
			}else{
				var pars = "point=" + point;
				par.push(pars);
			}

			var star = iframe.$('#star').val();
			if (star == '') {
				iframe.$('#startip').removeClass("onShow").addClass("onError").html('请输入星星数!');
				return false;
			}else{
				var pars = "star=" + star;
				par.push(pars);
			}

			if (iframe.$('#allowpost').is(":checked")) {
				var pars = "allowpost=1";
				par.push(pars);
			}else{
				var pars = "allowpost=0";
				par.push(pars);
			}

			if (iframe.$('#allowpostverify').is(":checked")) {
				var pars = "allowpostverify=1";
				par.push(pars);
			}else{
				var pars = "allowpostverify=0";
				par.push(pars);
			}

			if (iframe.$('#allowupgrade').is(":checked")) {
				var pars = "allowupgrade=1";
				par.push(pars);
			}else{
				var pars = "allowupgrade=0";
				par.push(pars);
			}

			if (iframe.$('#allowsendmessage').is(":checked")) {
				var pars = "allowsendmessage=1";
				par.push(pars);
			}else{
				var pars = "allowsendmessage=0";
				par.push(pars);
			}

			if (iframe.$('#allowattachment').is(":checked")) {
				var pars = "allowattachment=1";
				par.push(pars);
			}else{
				var pars = "allowattachment=0";
				par.push(pars);
			}

			if (iframe.$('#allowsearch').is(":checked")) {
				var pars = "allowsearch=1";
				par.push(pars);
			}else{
				var pars = "allowsearch=0";
				par.push(pars);
			}

			var priceday = iframe.$('#priceday').val();
			if (priceday == '') {
				var pars = "priceday=0";
				par.push(pars);
			}else{
				var pars = "priceday=" + priceday;
				par.push(pars);
			}

			var pricemonth = iframe.$('#pricemonth').val();
			if (pricemonth == '') {
				var pars = "pricemonth=0";
				par.push(pars);
			}else{
				var pars = "pricemonth=" + pricemonth;
				par.push(pars);
			}

			var priceyear = iframe.$('#priceyear').val();
			if (priceyear == '') {
				var pars = "priceyear=0";
				par.push(pars);
			}else{
				var pars = "priceyear=" + priceyear;
				par.push(pars);
			}

			var allowmessage = iframe.$('#allowmessage').val();
			if (allowmessage == '') {
				iframe.$('#allowmessagetip').removeClass("onShow").addClass("onError").html('请输入最大短消息数!');
				return false;
			}else{
				var pars = "allowmessage=" + allowmessage;
				par.push(pars);
			}

			var allowpostnum = iframe.$('#allowpostnum').val();
			if (allowpostnum == '') {
				iframe.$('#allowpostnumip').removeClass("onShow").addClass("onError").html('请输入日最大投稿数!');
				return false;
			}else{
				var pars = "allowpostnum=" + allowpostnum;
				par.push(pars);
			}

			var usernamecolor = iframe.$('#usernamecolor').val();
			var pars = "usernamecolor=" + usernamecolor;
			par.push(pars);

			var icon = iframe.$('#icon').val();
			var pars = "icon=" + icon;
			par.push(pars);

			var desc = iframe.$('#desc').val();
			var pars = "desc=" + desc;
			par.push(pars);

			var status = iframe.$("input[name='status']:checked").val();
			var pars = "status=" + status;
			par.push(pars);

			var pars = "csrf_token=" + csrf_token;
			par.push(pars);

			pars = par.join("&");

			$.ajax({
				type : "POST",
				url : "/Group/edit/",
				data : pars,
				success : function(tmp) {
					if (tmp.status == 1) {
						right_refresh();
						notice_tips("编辑成功!");
					} else {
						notice_tips(tmp.message);
					}
				}
			});
		},
		cancel : true
	});
}

/**
 * 删除用户组
 */
function del(id) {
	if (id == '') {
		notice_tips("参数错误!");
		return false;
	}

	art.dialog.confirm('你确定要删除这个用户吗?', function() {
		$.post("/Group/delete/",{'id':id,'csrf_token':csrf_token}, function(tmp){
			if (tmp.status == 1) {
				right_refresh();
				notice_tips("删除成功!");
			} else {
				notice_tips(tmp.content);
			}
		});

	}, function() {
		notice_tips("你取消了删除用户操作!");
	});
}

function set_username_color(color) {
	$('#usernamecolor').val(color);
}